sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -2107,8 +2107,8 @@ case class ParseToDate(
 
   override lazy val replacement: Expression = withOrigin(origin) {
     format.map { f =>
-      Cast(GetTimestamp(left, f, TimestampType, "try_to_date", timeZoneId, ansiEnabled), DateType,
-        timeZoneId, EvalMode.fromBoolean(ansiEnabled))
+      Cast(GetTimestamp(left, f, TimestampType, "try_to_timestamp", timeZoneId, ansiEnabled),
```

Comment:
In which cases, we pass a value to `GetTimestamp` different from "try_to_timestamp"?